fastplotlib/plot.py

```diff
@@ -103,4 +107,54 @@ def show(self, autoscale: bool = True):
         if autoscale:
             self.auto_scale(maintain_aspect=True, zoom=0.95)
 
-        return self.canvas
+        if self.toolbar:
```

Comment:
Just make this a kwarg in show() instead of the constructor. Create a new toolbar widget instance in show() it it's not instantiated, else use existing instance.